source/base/multithread_info.cc

```diff
@@ -213,5 +208,15 @@ unsigned int MultithreadInfo::n_max_threads = numbers::invalid_unsigned_int;
 // definition of the variable which is declared `extern' in the .h file
 MultithreadInfo multithread_info;
 
+// Force the first call to set_thread_limit happen before any tasks in TBB are
+// used. This is necessary as tbb::task_scheduler_init has no effect if TBB
+// got automatically initialized (which happens the first time we use it).
+struct DoOnce
+{
+  DoOnce ()
+  {
+    MultithreadInfo::set_thread_limit (numbers::invalid_unsigned_int);
+  }
+} do_once;
```

Comment:
You don't want this variable to be exported from this object file (since its name might clash with other symbols elsewhere). To avoid this, you can either put it into an anonymous namespace, or mark it as `static`.